File: src/parser/cpp/source/issue.cpp
#include <string>

#include <termcolor.hpp>  // [https://github.com/ikalnytskyi/termcolor]

using namespace std;

void issue_hint(const string &filename, int line, int col, const string &message) {
	string itype = "\033[32;1mHint:\033[0m";
	string fileinfo = "\033[1m" + filename + "(" + to_string(line) + ", " + to_string(col) + ")\033[0m";

	cout << fileinfo << " " << itype << " " << message << endl;
}

void issue_warn(const string &filename, int line, int col, const string &message) {
	string itype = "\033[33;1mWarning:\033[0m";
	string fileinfo = "\033[1m" + filename + "(" + to_string(line) + ", " + to_string(col) + ")\033[0m";

	cout << fileinfo << " " << itype << " " << message << endl;
}

void issue_error(const string &filename, int line, int col, const string &message) {
	string itype = "\033[31;1mError:\033[0m";
	string fileinfo = "\033[1m" + filename + "(" + to_string(line) + ", " + to_string(col) + ")\033[0m";

	cout << fileinfo << " " << itype << " " << message << endl;
	exit(1);
}

File: src/parser/nim2/helpers/issue.nim
import ../utils/[chalk]

# \033: [https://stackoverflow.com/a/10170631]

proc issue_hint*(filename: string, line: int, col: int, message: string) =
    let itype = "Hint:".chalk("green", "bold")
    let fileinfo = (filename & "(" & $line & ", " & $col & ")").chalk("bold")

    echo fileinfo & " " & itype & " " & message

proc issue_warn*(filename: string, line: int, col: int, message: string) =
    let itype = "Warning:".chalk("yellow", "bold")
    let fileinfo = (filename & "(" & $line & ", " & $col & ")").chalk("bold")

    echo fileinfo & " " & itype & " " & message

proc issue_error*(filename: string, line: int, col: int, message: string) =
    let itype = "Error:".chalk("red", "bold")
    let fileinfo = (filename & "(" & $line & ", " & $col & ")").chalk("bold")

    echo fileinfo & " " & itype & " " & message
    quit()
